tictactoe.js

```diff
@@ -0,0 +1,336 @@
+            if (curPlayer === 'X' && ((ttTable[0] === 'X' && ttTable[1] === 'X' && ttTable[2] === 'X') || (ttTable[3] === 'X' && ttTable[4] === 'X' && ttTable[5] === 'X') || (ttTable[6] === 'X' && ttTable[7] === 'X' && ttTable[8] === 'X') || (ttTable[0] === 'X' && ttTable[3] === 'X' && ttTable[6] === 'X') || (ttTable[1] === 'X' && ttTable[4] === 'X' && ttTable[7] === 'X') || (ttTable[2] === 'X' && ttTable[5] === 'X' && ttTable[8] === 'X') || (ttTable[0] === 'X' && ttTable[4] === 'X' && ttTable[8] === 'X') || (ttTable[2] === 'X' && ttTable[4] === 'X' && ttTable[6] === 'X'))) {
+                console.log('\n'
+                    + ' ' + ttTable[0] + ' | ' + ttTable[1] + ' | ' + ttTable[2] + ' \n'
+                    + '--- --- ---\n'
+                    + ' ' + ttTable[3] + ' | ' + ttTable[4] + ' | ' + ttTable[5] + ' \n'
+                    + '--- --- ---\n'
+                    + ' ' + ttTable[6] + ' | ' + ttTable[7] + ' | ' + ttTable[8] + ' \n');
+                console.log('\nPlayer X has won, game over!\n');
+                process.exit();
+                // Winner is O
+            } else if (curPlayer === 'O' && ((ttTable[0] === 'O' && ttTable[1] === 'O' && ttTable[2] === 'O') || (ttTable[3] === 'O' && ttTable[4] === 'O' && ttTable[5] === 'O') || (ttTable[6] === 'O' && ttTable[7] === 'O' && ttTable[8] === 'O') || (ttTable[0] === 'O' && ttTable[3] === 'O' && ttTable[6] === 'O') || (ttTable[1] === 'O' && ttTable[4] === 'O' && ttTable[7] === 'O') || (ttTable[2] === 'O' && ttTable[5] === 'O' && ttTable[8] === 'O') || (ttTable[0] === 'O' && ttTable[4] === 'O' && ttTable[8] === 'O') || (ttTable[2] === 'O' && ttTable[4] === 'O' && ttTable[6] === 'O'))) {
+                console.log('\n'
+                    + ' ' + ttTable[0] + ' | ' + ttTable[1] + ' | ' + ttTable[2] + ' \n'
+                    + '--- --- ---\n'
+                    + ' ' + ttTable[3] + ' | ' + ttTable[4] + ' | ' + ttTable[5] + ' \n'
+                    + '--- --- ---\n'
+                    + ' ' + ttTable[6] + ' | ' + ttTable[7] + ' | ' + ttTable[8] + ' \n');
+                console.log('\nPlayer O has won, game over!\n');
+                process.exit();
+            }else if (ttTable[0] !='1' && ttTable[1] != '2' && ttTable[2] != '3' && ttTable[3] != '4' && ttTable[4] != '5' && ttTable[5] != '6' &&  ttTable[6] != '7' && ttTable[7] != '8' && ttTable[8] != '9')  {
+                console.log('\n'
+                    + ' ' + ttTable[0] + ' | ' + ttTable[1] + ' | ' + ttTable[2] + ' \n'
+                    + '--- --- ---\n'
+                    + ' ' + ttTable[3] + ' | ' + ttTable[4] + ' | ' + ttTable[5] + ' \n'
+                    + '--- --- ---\n'
+                    + ' ' + ttTable[6] + ' | ' + ttTable[7] + ' | ' + ttTable[8] + ' \n');
+                console.log('\nThere are no winners!\n');
+                process.exit();
+            }    else {
+                //           return start()
+                return player(nextPlayer, xPlay, oPlay);
```

Comment:
Recursion is cool, but in cases like this it makes the code harder to understand since the game is not really recursive, but alternating until a condition is reached. Instead of doing `return player()` try just `return winningPlayer` and let a separate higher-level function loop and call `player()` and `whoWon()` on its own. That way each individual function is responsible for a smaller set of tasks.

something like...

```
let gameOver = false;
let currentPlayer = 'X';
while (!gameOver) { 
  play(currentPlayer, ttTable)
  let winner = whoWon(ttTable)
  if (winner === 'X' || winner === 'O' || winner = 'S') {
    gameOver = true;
  }
}
```